server/repository/src/db_diesel/requisition_line/requisition_line_row.rs

```diff
@@ -69,6 +69,23 @@ impl<'a> RequisitionLineRowRepository<'a> {
         self.insert_changelog(row, RowActionType::Upsert)
     }
 
+    pub fn update_approved_quantity_by_item_id(
+        &self,
+        requisition_id: &str,
+        item_id: &str,
+        approved_quantity: f64,
+    ) -> Result<(), RepositoryError> {
+        diesel::update(requisition_line_dsl::requisition_line)
+            .filter(
+                requisition_line_dsl::requisition_id
+                    .eq(requisition_id)
+                    .and(requisition_line_dsl::item_link_id.eq(item_id)),
+            )
+            .set(requisition_line_dsl::approved_quantity.eq(approved_quantity))
+            .execute(self.connection.lock().connection())?;
+        Ok(())
```

Comment:
Should this insert a changelog? Initial thought was yes (it's a change, we want it to sync) but a lot of these field updater methods don't seem to call insert_changelog? Was that because the old changelog triggers used to capture these changes, and we missed these field updater methods in the update to manage changelogs in Rust? Or do they not add changelogs on purpose?

Reply:
Some of the field updaters are coupled together with the normal upserts, so the changelogs are created for the upsert, and when sync runs it catches the field updates too... maybe we should add it in as a fail safe? 🤔 Should definitely be adding changelog if the fields are being updated independently tho...

Reply:
Woah is that an explicit design decision, or just how it's happened? Feels like a pretty big risk that the field updater could be used somewhere else and the change not captured? Like the repository layer can't know whether its being used independently or after a regular upsert 👀

Reply:
Was missed during changelog changes.... can create a new issue for it (: